### PR TITLE
[dav1d] Add dav1d AV1 decoder port

### PR DESCRIPTION
**Pull request by RichLogan**
Adds `dav1d` AV1 decoder port

- Which triplets are supported/not supported? Have you updated the CI baseline?

Struggling with MSVC underscore prefixes on x86-windows. Disabled support in manifest for now.

### CODE REVIEW

**Comment by RichLogan** [comment]
I think that these are failing on linux due a bug in upstream meson. I will submit a PR to rev vcpkg's meson to try and resolve this.

File: ports/dav1d/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "dav1d",
+  "version-string": "0.7.1",
+  "port-version": 1,
```
Comment (NancyLi1013) [inline ports/dav1d/vcpkg.json:4]:
```suggestion
```
For new port or update to new version, there is unnecessary to add this field. Or you can set it as `0`.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
@RichLogan 

Could you please sign CLA first?

**Comment by RichLogan** [comment]
Depends: https://github.com/microsoft/vcpkg/pull/15200

**Comment by JackBoosY** [comment]
Can you continue this PR now?

**Comment by RichLogan** [comment]
I think this is ready but I need some help with why x86-windows is failing on the "Validate Version Files" step.

**Comment by NancyLi1013** [comment]
Could you please merge master and run command `vcpkg x-add-version dav1d
`?
```
Found the following errors:
Error: While validating port dav1d.
       Missing expected versions file at: C:\a\1\s\port_versions\d-\dav1d.json
       Run:

           vcpkg x-add-version dav1d

       to create the versions file.

To attempt to resolve all errors at once, run:

    vcpkg x-add-version --all
```

**Comment by RichLogan** [comment]
@NancyLi1013 All done, thanks!

File: ports/dav1d/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    OPTIONS
+        --default-library=${LIBRARY_TYPE}
+        -Denable_tests=false
+        -Denable_tools=false
+)
+
+vcpkg_install_meson()
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
```
Comment (NancyLi1013) [inline ports/dav1d/portfile.cmake:31]: Why do you remove these folders?

File: ports/dav1d/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,32 @@
+vcpkg_from_gitlab(
```
Comment (NancyLi1013) [inline ports/dav1d/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm" "x86" ON_TARGET "uwp")

vcpkg_from_gitlab(
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/dav1d/portfile.cmake (hunk 2 above)
Comment (RichLogan) [reply ports/dav1d/portfile.cmake:31]: Good spot this was prior to properly disabling tools with an option

**Comment by RichLogan** [comment]
@NancyLi1013 Can this be merged now?

**Review by JackBoosY — APPROVED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @RichLogan.

**Pull request merged**